sourcegraph/cody
# Onboarding: New login experience behind QA flag

This adds a simplified login interface for easier onboarding.

Caveats:
- No automated tests yet.
- No experiment selection or caching yet, noted with TODOs.
- No exposure logging yet, noted with TODOs.

## Test plan

Check nothing broke:

```
pnpm test:unittest && pnpm test:e2e && pnpm test:integration
```

Will add automated tests for this new UI in a follow-up change.

https://www.loom.com/share/3660f9da9e7f45b8b6fb4032ac91360f?sid=3db70fde-fe8e-4c3a-a36e-bf14d47406d4

Manual test:

1. Go to User Settings (JSON)
2. Set a key `cody.todo-internal-onboarding-qa` to any value.
3. Run the extension. Sign out if necessary new sign in UI should appear.

For each of GitHub, GitLab, Google:

1. Open a browser and sign out of sourcegraph.com and the provider
2. Click the sign in button in the extension
3. Sign in to the provider
4. Complete the post-sign up survey, if necessary.
5. Open the VScode URL and you should be logged in.
6. Log out. The new login experience should appear.

Repeat, but be signed *in* to the identity provider already.

Repeat all of the above with different default system browsers, operating systems, etc.

Click the Sourcegraph Enterprise sign-in link and confirm those sign in options continue to work.
